### DISCUSSION TITLE
Is it possible to make an UI without using XML?

### DISCUSSION THREAD

**Discussion by ghost** [SpartanJ/eepp#80] [General]
There is one thing I want to clarify: I'm not going to use this library. This question only stems from my curiosity. The XML on README dreaded me. Is using XML the only way to make an UI?

**Comment by SpartanJ** [comment]
No, XML is just an abstraction and it's optional, you can just build everything from code, it looks like [this](https://github.com/SpartanJ/eepp?tab=readme-ov-file#ui-widgets-with-c-example) or [a more complete example](https://github.com/SpartanJ/eepp/blob/develop/src/eepp/ui/uifiledialog.cpp).
